src/uhs/atomizer/atomizer/state_machine.cpp

```diff
@@ -203,7 +203,6 @@ namespace cbdc::atomizer {
     void state_machine::create_snapshot(
         nuraft::snapshot& s,
         nuraft::async_result<bool>::handler_type& when_done) {
-        assert(s.get_last_log_idx() == last_commit_index());
```

Comment:
I believe this `assert` to be a bug. However, the a different assert may make sense depending on the intent of the original author. @metalicjames - please comment.

Reply:
The assertion is there because it's an assumption the rest of the method makes. `create_snapshot` only ever creates a snapshot as of `last_commit_index()`. Thus, if NuRaft puts something else in `s.get_last_log_idx()`, the generated snapshot will be for the incorrect log index. It wasn't super clear, but I assumed NuRaft would only ever call this method with `s.get_last_log_idx() == last_commit_index()`. If that's not true, we need to edit this method to support generating a snapshot at any log index rather than the current one.

Reply:
@metalicjames : please see #125